modules/masterbar/admin-menu/class-wpcom-admin-menu.php

```diff
@@ -0,0 +1,322 @@
+<?php
+/**
+ * WP.com Admin Menu file.
+ *
+ * @package Jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+use Automattic\Jetpack\Status;
+
+/**
+ * Class WPcom_Admin_Menu.
+ */
+class WPcom_Admin_Menu extends Admin_Menu {
+
+	/**
+	 * Sets up class properties for REST API requests.
+	 */
+	public function rest_api_init() {
+		parent::rest_api_init();
+
+		// Get domain for requested site.
+		$this->domain = ( new Status() )->get_site_suffix();
+	}
+
+	/**
+	 * Create the desired menu output.
+	 */
+	public function reregister_menu_items() {
```

Comment:
It looks like `reregister_menu_items` for atomic and WPCOM classes are the same. Do you think we can move it into a parent class?

Reply:
This is something I have also pointed here https://github.com/Automattic/jetpack/pull/18105#pullrequestreview-557078033

I am not sure what @obenland had in mind when duplicating this methods, Konstantin any hints? 

Since this PR is now tested and enormous, I would suggest merging it and following up with merging code.